amy/dashboard/views.py

```diff
@@ -229,7 +234,33 @@ def training_progress(request):
     progress_welcome = get_passed_or_last_progress(request.user, "Welcome Session")
     progress_demo = get_passed_or_last_progress(request.user, "Demo")
 
-    if request.method == "POST":
+    context = {
+        "title": "Your training progress",
+        "progress_training": progress_training,
+        "progress_get_involved": progress_get_involved,
+        "progress_welcome": progress_welcome,
+        "progress_demo": progress_demo,
+    }
+    return render(request, "dashboard/training_progress.html", context)
+
+
+class GetInvolvedCreateView(LoginRequiredMixin, RedirectSupportMixin, AMYCreateView):
+    # permission_required = TODO
+    model = TrainingProgress
+    form_class = GetInvolvedForm
+    template_name = "get_involved_form.html"
+    success_url = reverse_lazy("training-progress")
+    success_message = (
+        "Thank you. Your Get Involved submission will be reviewed within 7-10 days."
+    )
+
+    def get_context_data(self, **kwargs) -> dict:
+        context = super().get_context_data(**kwargs)
+        context["title"] = str("Submit your Get Involved activity")
+        return context
+
+    def post(self, request):
+        self.object = None
```

Comment:
Not sure what this line is for here, can you explain? (If it's still needed after moving to `get_form_kwargs`)

Reply:
It's no longer needed, but it was an addition to get the call to either `form_valid` or `form_invalid` to work.